chrome/content/zotero/elements/notesContext.js

```diff
@@ -270,66 +292,82 @@
-			let parentTitleContainer;
-			let vbox;
+			let editor;
+
 			if (isChild) {
-				vbox = document.createXULElement('vbox');
+				let vbox = document.createXULElement('vbox');
 				vbox.setAttribute('data-tab-id', Zotero_Tabs.selectedID);
 				vbox.style.display = 'flex';
 
 				editor = new (customElements.get('note-editor'));
 				editor.style.flex = "1";
-				vbox.append(editor);
 
-				tabNotesDeck.append(vbox);
+				vbox.append(editor);
 
-				editor.mode = this.editable ? 'edit' : 'view';
-				editor.item = item;
-				editor.parentItem = null;
+				this.tabNotesDeck.append(vbox);
 
 				this.mode = "childNote";
-				tabNotesDeck.setAttribute('selectedIndex', tabNotesDeck.children.length - 1);
-
-				parentTitleContainer = this.querySelector('.context-note-child > .zotero-context-pane-editor-parent-line');
+				this.tabNotesDeck.selectedIndex = this.tabNotesDeck.children.length - 1;
 			}
 			else {
-				this.mode = "standaloneNote";
-				editor.mode = this.editable ? 'edit' : 'view';
-				editor.item = item;
-				editor.parentItem = null;
+				// Try to reuse existing editor
+				if (this.standaloneEditor) {
+					editor = this.standaloneEditor;
+				}
+				else {
+					editor = new (customElements.get('note-editor'));
+					editor.classList.add("zotero-context-pane-pinned-note");
+					editor.style.flex = "1";
 
-				parentTitleContainer = node.querySelector('.context-note-standalone > .zotero-context-pane-editor-parent-line');
+					this.standaloneNoteContainer.append(editor);
+					this.standaloneEditor = editor;
+				}
+				this.mode = "standaloneNote";
 			}
 
+			editor.mode = this.editable ? 'edit' : 'view';
+			editor.item = item;
+			editor.parentItem = null;
 			editor.focus();
 
+			this.updatePinnedNoteTitle();
+			ZoteroContextPane.updateAddToNote();
+		}
+
+		updatePinnedNoteTitle() {
+			let item = this._getCurrentEditor()?.item;
+			let title = this.selectedPanel?.querySelector('.parent-title');
 			let parentItem = item.parentItem;
-			let container = document.createElement('div');
-			container.classList.add("parent-title-container");
-			let returnBtn = document.createXULElement("toolbarbutton");
-			returnBtn.classList.add("zotero-tb-note-return");
-			returnBtn.addEventListener("command", () => {
-				// Immediately save note content before vbox with note-editor iframe is destroyed below
-				editor.saveSync();
-				ZoteroContextPane.context.mode = "notes";
-				this.mode = "notesList";
-				vbox?.remove();
-				ZoteroContextPane.updateAddToNote();
-				this._preventViewTypeCache = true;
-			});
-			let title = document.createElement('div');
-			title.className = 'parent-title';
 			title.textContent = parentItem?.getDisplayTitle() || '';
-			container.append(returnBtn, title);
-			parentTitleContainer.replaceChildren(container);
-			ZoteroContextPane.updateAddToNote();
 		}
 
 		updateNotesListFromCache() {
 			this._updateNotesList(true);
 		}
 
+
```

Comment:
5. Make the logic of deciding what to show in the notes context (standalone note, current item note, or notes list) clearer and some bug fixes